src/components/dashboard/dashboard.spec.ts

```diff
@@ -0,0 +1,277 @@
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [DashboardModule],
+            declarations: [DashboardComponent]
+        })
+            .compileComponents();
+    }));
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(DashboardComponent);
+        component = fixture.componentInstance;
+        nativeElement = fixture.nativeElement;
+        widget = nativeElement.querySelector('ux-dashboard-widget');
+        widgetC = nativeElement.querySelector('#widget-C');
+        fixture.detectChanges();
+    });
+
+    it('should initialise correctly', () => {
+        expect(component).toBeTruthy();
+    });
+
+    it('should reposition widgets if columns option adjusted', async() => {
+        let adjustedOptions: DashboardOptions  = {
+            columns: 3,
+            padding: 5,
+            rowHeight: 400,
+            emptyRow: false,
+            minWidth: 187
+        };
+        component.options = component.initialOptions;
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        expect(widgetC.style.top).toBe('400px');
+
+        component.options = adjustedOptions;
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        expect(widgetC.style.top).toBe('800px');
+    });
+
+    it('should increase widget padding if padding option increased', async() => {
+        let adjustedOptions: DashboardOptions  = {
+            columns: 6,
+            padding: 10,
+            rowHeight: 200,
+            emptyRow: false,
+            minWidth: 187
+        };
+        expect(widget.style.padding).toBe('5px');
+
+        component.options = adjustedOptions;
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        expect(widget.style.padding).toBe('10px');
+    });
+
+    it('should increase widget rowHeight if rowHeight option increased', async() => {
+        let adjustedOptions: DashboardOptions  = {
+            columns: 6,
+            padding: 10,
+            rowHeight: 400,
+            emptyRow: false,
+            minWidth: 187
+        };
+
+        component.options = component.initialOptions;
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        expect(widget.style.height).toBe('200px');
+
+        component.options = adjustedOptions;
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        expect(widget.style.height).toBe('400px');
+    });
+
+    it('should set widget rowHeight to equal options minWidth if rowHeight less than minWidth', async() => {
+        let adjustedOptions: DashboardOptions  = {
+            columns: 6,
+            padding: 10,
+            rowHeight: 200,
+            emptyRow: false,
+            minWidth: 400
+        };
+
+        component.options = adjustedOptions;
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        expect(widget.style.height).toBe('400px');
```

Comment:
The purpose of minWidth is to switch the dashboard to stacked mode, so if you set the test up to have e.g. two widgets on the first row and one on the next, you can test that increasing the minWidth causes the dashboard height to increase from 2*rowHeight to 3*rowHeight.